### PR TITLE
Move around references

### PR DESCRIPTION
**Pull request by benarent**
This PR moves around some reference files, for a long time we've had a monster yaml file in the admin manual. This has resulted a few users getting into a pickle, as they copy over the whole thing. I've added a smaller reference file based on `teleport configure`.  

This adds a short WIP section for the API work, I will likely move it elsewhere, but wanted a place to start.

### CODE REVIEW

File: docs/4.3/admin-guide.md [hunk 1]
```diff
@@ -126,7 +126,10 @@ Docs](cli-docs.md#teleport-start) or run `teleport start --help`
 ### Configuration File
 
 Teleport uses the YAML file format for configuration. A sample configuration
-file is shown below. By default, it is stored in `/etc/teleport.yaml`
+file is shown below. By default, it is stored in `/etc/teleport.yaml`, below is
+an expanded an commented version from `teleport configure`.
```
Comment (webvictim) [inline docs/4.3/admin-guide.md:130]: `expanded an commented` -> `expanded and commented`

File: docs/4.3/admin-guide.md [hunk 2]
```diff
@@ -135,315 +138,102 @@ file is shown below. By default, it is stored in `/etc/teleport.yaml`
...
-        # This setting is not required if the Teleport proxy service is
-        # deployed inside a Kubernetes cluster. Otherwise, Teleport proxy
-        # will use the credentials from this file:
-        kubeconfig_file: /path/to/kube/config
+  enabled: "yes"
+  listen_addr: 0.0.0.0:3023
+  web_listen_addr: 0.0.0.0:3080
+  tunnel_listen_addr: 0.0.0.0:3024
+
+  # The DNS name the proxy HTTPS endpoint as accessible by cluster users.
```
Comment (webvictim) [inline docs/4.3/admin-guide.md:227]: `name the proxy` -> `name of the proxy`

File: docs/4.3/config_reference.md [hunk 3]
```diff
@@ -0,0 +1,320 @@
+# Teleport Configuration Reference
+
+## teleport.yaml
+
+Teleport uses the YAML file format for configuration. A full configuration reference
+file is shown below, this provides comments and all avaible options for `teleport.yaml`
```
Comment (webvictim) [inline docs/4.3/config_reference.md:6]: `avaible` -> `available`

File: docs/4.3/config_reference.md [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+    # outbound (from behind the firewall) connection to this address.
+    # This will allow users of the outside CA to connect to behind-the-firewall
+    # nodes.
+    tunnel_listen_addr: 0.0.0.0:3024
+
+    # The HTTPS listen address to serve the Web UI and also to authenticate the
+    # command line (CLI) users via password+HOTP
+    web_listen_addr: 0.0.0.0:3080
+
+    # The DNS name the proxy HTTPS endpoint as accessible by cluster users.
```
Comment (webvictim) [inline docs/4.3/config_reference.md:286]: `name the proxy` -> `name of the proxy`

**Review by webvictim — COMMENTED** [review]
#gusreview

**Comment by benarent** [comment]
Thanks @webvictim  PTAL

**Review by webvictim — APPROVED** [review]

**Pull request merged**